test/web/Program.cs

```diff
@@ -1,41 +1,106 @@
 using System;
 using System.Collections.Generic;
-using System.Linq;
-using System.Threading.Tasks;
+using System.IO;
+using CommandLine;
 using Microsoft.AspNetCore.Hosting;
-using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.Hosting;
-using Microsoft.Extensions.Logging;
 using Tmds.LinuxAsync;
 
 namespace web
 {
+    public enum SocketEngineType
+    {
+        EPoll,
+        IOUring
+    }
+
+    public class CommandLineOptions
+    {
+        [Option('e', "engine", Required = false, Default = SocketEngineType.IOUring, HelpText = "EPoll/IOUring")]
+        public SocketEngineType SocketEngine { get; set; }
+
+        [Option('t', "thread-count", Required = false, Default = 1, HelpText = "Thread Count, default value is 1")]
+        public int ThreadCount { get; set; }
+
+        [Option('a', "aio", Required = false, Default = true, HelpText = "Use Linux AIO")]
+        public bool UseAio { get; set; }
+
+        [Option('c', "dispatch-continuations", Required = false, Default = false, HelpText = "Dispatch Continuations")]
+        public bool DispatchContinuations { get; set; }
+
+        [Option('s', "defer-sends", Required = false, Default = true, HelpText = "Defer Sends")]
+        public bool DeferSends { get; set; }
+
+        [Option('r', "defer-receives", Required = false, Default = true, HelpText = "Defer Receives")]
+        public bool DeferReceives { get; set; }
```

Comment:
Let's set these defaults to match the default Socket behavior:
dispatch-continuations: true
defer-sends: false
defer-receives: false